### PR TITLE
feat(CRDs): add basic CRD support to the dynamic client

### PR DESCRIPTION
**Pull request by silasbw**

### CODE REVIEW

**Comment by silasbw** [comment]
Missing features / loose ends documented as issues: https://github.com/godaddy/kubernetes-client/issues/216, https://github.com/godaddy/kubernetes-client/issues/215

File: examples/crontabs-crd.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      "kind": "CronTab", 
+      "plural": "crontabs", 
+      "singular": "crontab"
+    }
+  }, 
+  "apiVersion": "apiextensions.k8s.io/v1beta1", 
+  "metadata": {
+    "name": "crontabs.stable.example.com"
+  }
+}
```
Comment (gergelyke) [inline examples/crontabs-crd.json]: new line ☘️

File: examples/using-crds.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//
+// Use a Custom Resource Definition to extend the Kubernetes API and the client.
+//
+const Client = require('kubernetes-client').SyncClient
```
Comment (gergelyke) [inline examples/using-crds.js]:
```

File: examples/crontabs-crd.json (hunk 1 above)
Comment (silasbw) [reply examples/crontabs-crd.json]: :)  I call it AAFN (Add A F****** Newline).

Comment (jeffpearce) [reply examples/crontabs-crd.json]: 😱

**Pull request merged**